src/figuresClasses.ts

```diff
@@ -1,19 +1,105 @@
-export interface Figure {
+export enum Shape {
+  TriangleShape = 'triangle',
+  CircleShape = 'circle',
+  RectangleShape = 'rectangle',
+}
+
+export enum Color {
+  RedColor = 'red',
+  GreenColor = 'green',
+  BlueColor = 'blue',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle {
+  public shape = Shape.TriangleShape;
 
-}
+  aSide: number;
+
+  bSide: number;
+
+  cSide: number;
+
+  public color: Color;
+
+  constructor(
+    color: Color,
+    aSide: number,
+    bSide: number,
+    cSide: number,
+  ) {
+    this.aSide = aSide;
+    this.bSide = bSide;
+    this.cSide = cSide;
+    this.color = color;
+
+    if (aSide + bSide <= cSide
+    || bSide + cSide <= aSide
+    || cSide + aSide <= bSide) {
+      throw new Error('Wrong length of sides');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const halfPerimeter: number = (this.aSide + this.bSide + this.cSide) / 2;
```

Comment:
if your variables have a primitive type such as number, boolean, string, then you can avoid write type for these variables